packages/block-editor/src/components/list-view/block.js

```diff
@@ -104,8 +104,7 @@ function ListViewBlock( {
 
 	const selectEditorBlock = useCallback(
 		( event ) => {
-			event.stopPropagation();
-			selectBlock( clientId );
+			selectBlock( event, clientId );
```

Comment:
What's the background behind this change?

Just an FYI, there was a bug that someone noticed yesterday that's related to `selectEditorBlock` - https://github.com/WordPress/gutenberg/pull/38760.

Reply:
Thanks for the heads-up about that PR! I'll rebase this PR tomorrow and do some extra testing to see if I need to refactor a little more there to preserve that fix. One of the main changes in this PR is that we need to pass up the click and keyboard events to where we use the `updateBlockSelection` function, so that the `useBlockSelection` hook can factor in the type of event, whether the shift key was held, and whether it was a click or keyboard event.

Reply:
I've adjusted the logic slightly (added in a separate `selectDuplicatedBlock` callback so that `selectBlock` is called correctly), but the behaviour from that fix is now preserved.